--- a/Modules/_ctypes/callbacks.c	2026-02-03 07:32:20
+++ b/Modules/_ctypes/callbacks.c	2026-02-16 03:18:08
@@ -6,6 +6,9 @@
 // windows.h must be included before pycore internal headers
 #ifdef MS_WIN32
 #  include <windows.h>
+#endif
+#if defined(__APPLE__)
+#  include <TargetConditionals.h>
 #endif
 
 #include "pycore_call.h"          // _PyObject_CallNoArgs()
@@ -348,6 +351,12 @@
     Py_ssize_t nargs, i;
     ffi_abi cc;
 
+#if defined(__APPLE__) && defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+    PyErr_SetString(PyExc_NotImplementedError,
+                    "ctypes callbacks are not supported on iPhoneOS in this build");
+    return NULL;
+#endif
+
     assert(PyTuple_Check(converters));
     nargs = PyTuple_GET_SIZE(converters);
     p = CThunkObject_new(st, nargs);
